geode-core/src/main/java/org/apache/geode/internal/cache/map/RegionMapPut.java

```diff
@@ -214,6 +225,9 @@ private void setOldValueEvenIfFaultedOut() {
   protected void unsetOldValueForDelta() {
     OffHeapHelper.release(getOldValueForDelta());
     setOldValueForDelta(null);
+    if (this.overwritePutIfAbsent) {
```

Comment:
I think it reads better if you use the accessor method here: if (isOverwritePutIfAbsent())